### PR TITLE
New: Rule to find layout-thrashing in your code

### PR DESCRIPTION
**Pull request by jankhunter**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[X] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
It is a new rule that deals with the main rendering problem - layout-thrashing

**Is there anything you'd like reviewers to focus on?**
Focus on the rendering part, after fixing the issues shown by this rule

### CODE REVIEW

**Comment by platinumazure** [comment]
Thanks @jankhunter for the PR!

The team will need to evaluate the proposed rule to see if it fits in ESLint core before we can merge this in.

Personally, I lean slightly against including this in core, as it's browser-specific. (Yes, we have a few Node-specific and browser-specific rules already, but we've significantly raised the bar for new rules since those other rules were created, and now we don't like to add rules that aren't applicable for the vast majority of runtimes.)

If the team decides not to include this in core, please feel free to find or create a plugin with this rule. It would be awesome to have a plugin that focuses on web performance!

**Comment by gyandeeps** [comment]
Maybe here: https://github.com/amilajack/eslint-plugin-dom ?

File: conf/eslint-recommended.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ module.exports = {
         "jsx-quotes": "off",
         "key-spacing": "off",
         "keyword-spacing": "off",
+        "layout-thrashing": "error",
```
Comment (Aladdin-ADD) [inline conf/eslint-recommended.js]: I don't think we should enable it in `eslint-recommended` -- changing `eslint-recommended` would be a breaking change. see https://github.com/eslint/eslint#semantic-versioning-policy

Comment (jankhunter) [reply conf/eslint-recommended.js]: @Aladdin-ADD Dint know what should be done in eslint-recommended.js file. now i did the required changes..

**Comment by platinumazure** [comment]
I've taken a quick look at the implementation. You may wish to look at our [code path analysis docs](https://eslint.org/docs/developer-guide/code-path-analysis) and see if those events might allow you to simplify the implementation.

**Comment by kaicataldo** [comment]
Thank you for the contribution. I agree with @platinumazure and don't think this meets our strict criteria for a new rule in core. That being said, I'd love to see a web performance plugin if it doesn't already exist!

**Comment by jankhunter** [comment]
Thank you for considering!! Thank you @platinumazure @kaicataldo @gyandeeps for your comments. I've learnt a lot more about ESLint during Pull Request 😀

**Comment by platinumazure** [comment]
Closing this PR as it seems there is team consensus in favor of not adding this to core ESLint. Thanks @jankhunter for contributing and I hope you can land this in eslint-plugin-dom or similar!

**Pull request closed without merging**